### PR TITLE
Bugfix/missing metis neighbors

### PR DESCRIPTION
**Pull request by corbett5**

### CODE REVIEW

File: src/ArrayOfSets.hpp [hunk 1]
```diff
@@ -422,13 +422,15 @@ class ArrayOfSets : protected ArrayOfSetsView< T, INDEX_TYPE >
...
      * @brief Callback signaling that the size of the set has increased.
+     * @param [in] curPtr the current pointer to the array.
      * @param [in] nToAdd the increase in the size.
      * @note This method doesn't actually change the size, it just checks if the new size
      *       exceeds the capacity of the set and if so reserves more space.
      * @return a pointer to the sets values.
      */
     inline
-    T * incrementSize( INDEX_TYPE const nToAdd ) const LVARRAY_RESTRICT_THIS
+    T * incrementSize( T * const LVARRAY_UNUSED_ARG( curPtr ),
```
Comment (rrsettgast) [inline src/ArrayOfSets.hpp]: can you change `LVARRAY_UNUSED_ARG` to `LVARRAY_UNUSED_PARAM`?...
and why add this parameter if it us unused?

File: src/ArrayOfSetsView.hpp [hunk 2]
```diff
@@ -300,7 +300,7 @@ class ArrayOfSetsView : protected ArrayOfArraysView< T, INDEX_TYPE, std::is_cons
     ARRAYOFARRAYS_CHECK_BOUNDS( i );
 
     INDEX_TYPE const setSize = sizeOfSet( i );
-    T const * const setValues = (*this)[i];
+    T * const setValues = getSetValues( i );
```
Comment (rrsettgast) [inline src/ArrayOfSetsView.hpp]: How did this work before when `setValues` was a pointer to const?

File: src/ArrayOfSets.hpp (hunk 1 above)
Comment (corbett5) [reply src/ArrayOfSets.hpp]: It's used in the default `sortedArrayManipulation::CallBack::incrementSize`.

File: src/ArrayOfSetsView.hpp (hunk 2 above)
Comment (corbett5) [reply src/ArrayOfSetsView.hpp]: The `setValues` is used to determine the number of insertions that take place, then the `incrementSize` call back returns the pointer that is used to do the insertions.

**Pull request merged**